### DISCUSSION TITLE
Drop support for Python 2.7 and focus on supporting Python 3.8+

### DISCUSSION THREAD

**Discussion by ZeroCool940711** [Sygil-Dev/whoosh-reloaded#44] [General]
I want to open a discussion to talk about this topic before getting to work on it, since this change might require some work and is an important one. It's also not something that can be done in just a day or two and might affect the future of this repo and Whoosh in general. 

I think that we should remove all the code that is there mainly to support older versions of Python (2.7-3.6) and are just ways to do the same thing but that require special treatment on older versions for one reason or another. Specifically, I want to drop support for anything older than Python 3, if possible, I would like to focus on adding better support for Python 3.8-3.12 and up. We could not only reduce the lines of code on the repo by about 20-30% but also improve maintainability by a similar amount or even higher since the new tools we added are reporting a huge amount of issues on several different locations about code that could be improved. We could also make use of some of the features and security stuff available on Python 3.8+ that don't exist on Python 2.7.

@cclauss, What do you think about this?

**Comment by cclauss** [comment]
Full speed ahead.  Python 2 died 1,494 days ago on 1/1/2020 so I see no reason to continue supporting it.

The same goes for all versions of the Python core team https://devguide.python.org/versions because supporting 5 versions of CPython is enough effort and basing production systems on unsupported software makes one a target.

**Comment by ZeroCool940711** [comment]
Good. I will start to work on that then.
